xbmc/interfaces/json-rpc/PlayerOperations.cpp

```diff
@@ -153,32 +153,38 @@ JSONRPC_STATUS CPlayerOperations::GetItem(const std::string &method, ITransportL
       }
       else if (player == Video)
       {
-        if (!CVideoLibrary::FillFileItem(g_application.CurrentFile(), fileItem, parameterObject))
+        if (!CVideoLibrary::FillFileItem(fileItem->GetPath(), fileItem, parameterObject))
         {
+          // Fallback to item details held by GUI but preserve path
+          //! @todo  check this fallback still needed, can GUI item have more details than app item?
           const CVideoInfoTag *currentVideoTag = CServiceBroker::GetGUI()->GetInfoManager().GetCurrentMovieTag();
           if (currentVideoTag != NULL)
           {
             std::string originalLabel = fileItem->GetLabel();
+            std::string originalPath = fileItem->GetPath();
             fileItem->SetFromVideoInfoTag(*currentVideoTag);
             if (fileItem->GetLabel().empty())
               fileItem->SetLabel(originalLabel);
+            fileItem->SetPath(originalPath);  // Ensure path unchanged
           }
-          fileItem->SetPath(g_application.CurrentFileItem().GetPath());
         }
       }
       else
       {
-        if (!CAudioLibrary::FillFileItem(g_application.CurrentFile(), fileItem, parameterObject))
+        if (!CAudioLibrary::FillFileItem(fileItem->GetPath(), fileItem, parameterObject))
         {
+          // Fallback to item details held by GUI but preserve path
+          //! @todo  check this fallback still needed, can GUI item have more details than app item?
           const MUSIC_INFO::CMusicInfoTag *currentMusicTag = CServiceBroker::GetGUI()->GetInfoManager().GetCurrentSongTag();
           if (currentMusicTag != NULL)
           {
             std::string originalLabel = fileItem->GetLabel();
+            std::string originalPath = fileItem->GetPath();
             fileItem->SetFromMusicInfoTag(*currentMusicTag);
             if (fileItem->GetLabel().empty())
               fileItem->SetLabel(originalLabel);
+            fileItem->SetPath(originalPath);  // Ensure path unchanged
```

Comment:
this is wrong. path of an fileitem must never be changed because path is id.

Reply:
SetFromMusicInfoTag must not change the path. It it does, please correct.

Reply:
Path can be set by `SetFromMusicInfoTag` here https://github.com/xbmc/xbmc/blob/067538b50084397f021ecea02e52f0d3825adbba/xbmc/FileItem.cpp#L1670
But no I am not willing something as fundamental to change `SetFromMusicInfoTag` at this stage in the dev cycle